engine/admin/resource_constraints.md

```diff
@@ -38,16 +38,30 @@ for [Memory Resource Controller](https://www.kernel.org/doc/Documentation/cgroup
 
 ### `--memory-swap` details
 
-- If unset, and `--memory` is set, the container can use twice as much swap
-      as the `--memory` setting, if the host container has swap memory configured.
-      For instance, if `--memory="300m"` and `--memory-swap` is not set, the
-      container can use 300m of memory and 600m of swap.
-- If set to a positive integer,  and if both `--memory` and `--memory-swap`
-      are set, `--memory-swap` represents the total amount of memory and swap
-      that can be used, and `--memory` controls the amount used by non-swap
-      memory. So if `--memory="300m"` and `--memory-swap="1g"`, the container
-      can use 300m of memory and 700m (1g - 300m) swap.
-- If set to `-1` (the default), the container is allowed to use unlimited swap memory.
+`--memory-swap` is a modifier flag that only has meaning if `--memory` is also
+set. Swap is disk-based storage, so is usually less performant than memory.
+Its setting can have complicated effects:
+
+- If `--memory-swap` is set to a positive integer, then both `--memory` and
+  `--memory-swap` must be set. `--memory-swap` represents the total amount of
+  memory and swap that can be used, and `--memory` controls the amount used by
+  non-swap memory. So if `--memory="300m"` and `--memory-swap="1g"`, the
+  container can use 300m of memory and 700m (`1g - 300m`) swap.
+
+- If `--memory-swap` is unset, and `--memory` is set, the container can use as much swap
```

Comment:
will have to check but I thought it was 2 * (memory)

Reply:
I send a PR for fix this. https://github.com/docker/docker.github.io/pull/9340 

`MemorySwap` is `Memory` + `Swap`
```
	if hostConfig.Memory > 0 && hostConfig.MemorySwap == 0 {
		// By default, MemorySwap is set to twice the size of Memory.
		hostConfig.MemorySwap = hostConfig.Memory * 2
	}

```